Fix domain check

hakehuang opened the pull request:
for platforms that has the ipc feature, but does not configured for application like samples/subsys/ipc/openamp, the build will be filtered out. but if we enable the --prep-artifacts-for-testing, we need add protection to check the domains, as at this time it could be NULL.

fixes: #87163

sylvioalves committed:
samples: openamp: filter out esp32 appcpu core

ESP32xx appcpu board should only be built with procpu.
Stand alone AMP support makes no sense.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

hakehuang committed:
openamp: remote: change platform allow to filter

change paltform allow to filter in align with the main
core application

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
twister: runner: --prep-artifacts-for-testing protection

need add protection when check with sysbuild filter out
application, the domains is NULL

fixes: #87163

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

nashif commented:
before proceeding with this, take a look at 

https://github.com/zephyrproject-rtos/zephyr/actions/runs/13874864850


and the many failures resulting from https://github.com/zephyrproject-rtos/zephyr/pull/85997 and attempts to fix them https://github.com/zephyrproject-rtos/zephyr/pull/87139/files

hakehuang commented:
> and the many failures resulting from #85997 and attempts to fix them https://github.com/zephyrproject-rtos/zephyr/pull/87139/files

@nashif thanks a lot for this updates, and really sorry for the troubles. I guess the fixes in #87139 is not complete, as samples/subsys/ipc/ipc_service/static_vrings also has such issue. let me update with his fixes and add to others

hakehuang committed:
samples: ipc_service: remove fatal_error when platform not supported

remove the FATAL_ERROR message, and leave it just as a message.
also enhance filter for this case with "1pc0" anmd "1pc1"

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
@sylvioalves , do you mind if I cherry-pick your commit from your fix PR #87139?

hakehuang commented:
@nashif please help to review this PR, it is kind of blocker to our board testing. Thanks a lot.
